### PR TITLE
Allow to use custom reporters

### LINKED ISSUE #1: [Feature request / Question] Alternative way to notify end users about issues
Hi,

Thanks for developing this gem, I found this gem very useful.
To my mind log file is a good way to communicate issues. 
I wanted to improve the management of reported issues (like resolving, skipping, UI)
I was able to implement is by parsing a log file and storing them in database.
It would be nice to have something like [ActiveSupport::Notifications](https://guides.rubyonrails.org/active_support_instrumentation.html#creating-custom-events) for the reporting purpose.
In my case it will replace log file parsing (which is not scalable) and maybe will allow to ignore issues I'm not intended to fix (like some things in layout, that are common to many pages)

What do you think?

Thanks

### PR DESCRIPTION
**Pull request by fatkodima**
Fixes #1.

It should be used like
```ruby
class MyReporter
  def report(title, created_records)
    # do actual reporting
  end
end

ColumnsTrace.reporter = MyReporter.new
```

cc @andriy-baran Wdyt on the implementation?

### CODE REVIEW

**Comment by andriy-baran** [comment]
Looks great. Only thing to improve is documentation
Something like this:
```ruby 
class MyReporter
  def report(title, created_records)
    title # => 'controller#action'
    created_records # => [<#ColumnsTrace::CreatedRecord>]
    created_records.each do |record|
      record.model # class of AR model
      record.accessed_fields  # array of fields
      record.unused_fields # array
      record.backtrace # string
      record.record # AR model instance
    end
  end
end

File: lib/columns_trace/created_record.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      @backtrace = backtrace
+    end
+
+    def accessed_fields
+      record.accessed_fields
+    end
+
+    def unused_fields
+      # We need to store this into local variable, because `record.attributes`
+      # will access all attributes.
```
Comment (andriy-baran) [inline lib/columns_trace/created_record.rb:20]: Do we need to call `accessed_fields` before `unused_fields` to use it right, or order does not matter?

**Comment by fatkodima** [comment]
Thanks for review! Improved docs.

File: lib/columns_trace/created_record.rb (hunk 1 above)
Comment (fatkodima) [reply lib/columns_trace/created_record.rb:39]: It should not matter, I push a small fix for this to actually be the case.

**Pull request merged**